src/org/zaproxy/zap/extension/websocket/WebSocketAPI.java

```diff
@@ -228,6 +234,21 @@ public boolean onMessageFrame(int channelId, WebSocketMessage message) {
 
                 @Override
                 public void onStateChange(State state, WebSocketProxy proxy) {
+                    if (state != State.CLOSED) {
+                        return;
+                    }
+
+                    WebsocketEventConsumer consumer = evMap.remove(proxy.getChannelId());
+                    if (consumer == null) {
+                        return;
+                    }
+
+                    // TODO replace the loop with:
+                    // ZAP.getEventBus().unregisterConsumer(consumer);
+                    // once available in targeted ZAP version (2.8.0?).

```

Comment:
"Todo add version" instead of the version question?

Reply:
No big deal, just a thought...

Reply:
Maybe remove it? It doesn't really matter when it will be included.

Reply:
Your call. It's such a minor thing, I approved the PR anyway.

Reply:
Removed :)